vaadin-split-layout-flow-parent/vaadin-split-layout-flow/src/main/java/com/vaadin/flow/component/splitlayout/SplitLayout.java

```diff
@@ -337,18 +338,18 @@ public void removeAll() {
     public static class SplitterDragendEvent
             extends ComponentEvent<SplitLayout> {
 
-        private static final String PRIMARY_FLEX_BASIS = "element.querySelector('[slot=\"primary\"]').style.flexBasis";
-        private static final String SECONDARY_FLEX_BASIS = "element.querySelector('[slot=\"secondary\"]').style.flexBasis";
+        private static final String PRIMARY_FLEX_BASIS = "element.querySelector(':scope > [slot=\"primary\"]').style.flexBasis";
+        private static final String SECONDARY_FLEX_BASIS = "element.querySelector(':scope > [slot=\"secondary\"]').style.flexBasis";
```

Comment:
These are the meaningful changes of the PR to prefix the selector with `:scope >` so the direct child is returned instead of the first child matching the query.